tests/Doctrine/Tests/Common/Cache/MemcacheCacheTest.php

```diff
@@ -51,4 +51,34 @@ protected function _getCacheDriver()
         $driver->setMemcache($this->memcache);
         return $driver;
     }
+    
+    /**
+     * The following values get converted to FALSE if you cast them to a boolean.
+     * @see http://php.net/manual/en/types.comparisons.php
+     */
+    public function falseCastedValuesProvider()
+    {
+        return array(
+            array(false),
+            array(null),
+            array(array()),
+            array('0'),
+            array(0),
+            array(0.0),
+            array('')
+        );
+    }
+    
+    /**
+     * Check to see that, even if the user saves a value in memcache that can be interpreted as false,
+     * memcache will still recognize that it's there.
+     * @dataProvider falseCastedValuesProvider
+     */
+    public function testFalseCastedValues($value)
```

Comment:
@stof In my environment, I can't test most of the drivers because they rely on systems which I simply don't have available (like SQLite or Redis or Couchbase etc.).

If I add these tests to the common test, then I think on a lot of these drivers the tests will fail. And I don't know how to fix them in those cases. Which means my PR will probably not be accepted.

If, however, you think that it will be ok, then I will gladly do the change because I agree with you: the implementation/behaviour should be consistent across all drivers.